cmd/src/users_verify.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"text/template"
+)
+
+const SubCommand = "verify-emails"
+
+// input = `[]VerifyEmailJson`
+// be careful if you change the vars format, since the vars need to match below
+const setEmailVerifiedTemplate = `
+mutation VerifyUserEmails(
+{{- range $idx, $_ := . -}}
+$user{{ printf "%09d" $idx }} ID!, $email{{ printf "%09d" $idx }} String!, 
+{{- end -}}
+) {
+{{- range $idx, $_ := . }}
+  verify{{ printf "%09d" $idx }}: setUserEmailVerified(user: $user{{ printf "%09d" $idx }}, email: $email{{ printf "%09d" $idx }}, verified: true) {
+    alwaysNil
+  }
+{{- end }}
+}
+`
+
+type VerifyEmailJson struct {
+	ID    string `json:"ID"`
+	Email string `json:"Email"`
+}
+
+func init() {
+	usage := `
+Examples:
+
+  Verify one or more user account(s):
+
+    	$ src users ` + SubCommand + ` -verify-json='[{"ID": "ID1", "Email": "email1"},...]'
```

Comment:
I apologize for not detailing this in the original issue, but in general with this CLI tool we prefer pipe-based commands over things like passing in JSON because it enables more powerful composition.

To be clear, I think the way this should work is:

Verify a single user's email:

```
src users emails verify -id='USERID'
```

Verify users en-mass:

```bash
src users list -query='myquery' -f='{{.ID}}' | xargs -n 1 -I USERID src users emails verify -id='USERID'
```

This matches the same way we e.g. tell users they can do mass user deletion:

https://github.com/sourcegraph/src-cli/blob/da78eacc42769eecb3d0b3789999dbeafdbff79b/cmd/src/users_delete.go#L20-L22

Reply:
This also simplifies our own code here and means e.g. you don't need to use the GraphQL template composition stuff here

Reply:
but AIUI that would also mean 250 round-trips to the API for verifying 250 users, right? OT1H, CPU and network are conceptually cheap, OTOH, each one of those requests is occupying a Sourcegraph instance that could be doing something else fun with those sockets.

I actually did strongly consider something akin to `src user emails verify -stdin` and letting them pipe JSON to stdin, but I didn't feel like pressing the issue

I'm \[unfortunately\] just a bystander now, so if you'd prefer that then I'll make it happen, but it feels hacky

Reply:
Separately, just using `-id` is insufficient because a user can have multiple emails; now, don't get me wrong, maybe verifying them all cart blanc is the 99% most often case, but without the ability to specify the explicit email of the UID, you're going to jam up someone's process